### ISSUE TITLE
How to enable IGPU for Metal compute?

### ISSUE THREAD

**Issue by diskdude** [osy86/HaC-Mini#213]
I need to use the IGPU for Metal compute, separately from the dGPU.

It looks like the default install has the IGPU disabled via the boot-args `-disablegfxfirmware`.   I've removed this, but the IGPU still doesn't show up as a Metal device, nor visible in System Report.  (I have the Intel graphics enabled in the BIOS.)

I see WEG listed in `ioreg`, but it's not matched?

```
    | +-o WhateverGreen  <class WhateverGreen, id 0x100000128, !registered, !matched, active, busy 0, retain 4>
```

Any tips to get the IGPU working?  I have seen that #12 references problems with wakeup, but with my 4K+2K+2K multi-monitor graphics I can't use sleep anyway #7.

- HaC Mini version: 2.6
- OSX version: 10.15.3
- NUC model: NUC8i7HVK
- Boot SSD model and protocol: EVO 970, m.2
- Display port(s) used: 2xUSB-C, 1xDP
- Display resolution(s): 4K, 2K, 2K
- Any TB3 devices: none
- Any USB devices: N/A

**Comment by osy86** [comment]
The flag doesn’t remove it. By default it’s enabled.

**Comment by diskdude** [comment]
OK, I'll put that boot-arg back in.  My apologies.

Any other ideas as to why it doesn't show up?  (I've also looked at `MTLCopyAllDevices()` and only the dGPU is listed.)

**Comment by osy86** [comment]
Yeah it’s the same behavior as Mac mini. The igpu is “offline”. You can force it to show up in a few ways but I can’t remember how exactly. Try using the amd gpu option when running the hac mini installer.

**Comment by diskdude** [comment]
> Try using the amd gpu option when running the hac mini installer.

It looks like that replaces the graphics-related boot-args to `shikigva=80 -radcodec`.   Unfortunately, that didn't help.

I've learned that [WEG defines an "offline" IGPU as one having no connectors](https://github.com/acidanthera/WhateverGreen/blob/master/Manual/FAQ.IntelHD.en.md):

> If you use a discrete graphics card (AMD or NVIDIA), having integrated Intel graphics enabled is still useful, as it can be used in offline mode (also known as "empty framebuffer", 0 connectors) for hardware encoding and decoding of media files and so forth.

The default HaC-Mini config uses the `ig-platform-id` of 0x59120003, which is an "offline" configuration with zero connected displays (zero connectors).  That looks good to me.

I loaded a debug version of WEG and used the `-wegdbg` boot-arg, but didn't see anything obviously wrong on a verbose boot. (I used a camera to catch the output.  Is there a way to look at that in macOS? `dmesg` doesn't go that far back.)

iStat menu sees the UHD630, but the Metal Framework does not list it as an device, nor does it appear System Information under Hardware > Graphics/Displays.  I do see it in IORegistryExplorer, along with an associated IntelAccelerator.

Any other suggestions? :)
